### PR TITLE
issue: 824880 Add Google Test based infrastructure

### PR DESCRIPTION
**Pull request by igor-ivanov**
Current gtest suite includes tcp, udp and vma suites:
- tcp socket, bind and event test cases
- vma ring
- udp none

How to build:
make tests

How to run:
This test should use two different NICs on the same node
`--addr=<server ip>:<client ip>:<any remote ip>`

usage help:
`./gtest --help`
list of tests:
`./gtest --gtest_list_tests`
run all tests:
`LD_PRELOAD=<path>/libvma.so ./gtest --addr=4.4.4.16:1.1.1.16:4.4.4.15`
run specific test:
`LD_PRELOAD=<path>/libvma.so ./gtest --addr=4.4.4.16:1.1.1.16:4.4.4.15 --gtest_filter=tcp_bind.ti_1`

@OphirMunk @rosenbaumalex please look at this.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/472/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/456/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/458/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/473/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@igor-ivanov is there a reason NOT to squash commits?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/464/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk I prefer to have separate commits

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/481/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
:+1:

File: tests/latency_test/tcp_lat.cpp [hunk 1]
```diff
@@ -722,6 +722,7 @@ int main(int argc, char *argv[])
 	int poll_mode = 0;
 	int testn = 1;
 
+	(void)poll_mode;
```
Comment (OphirMunk) [inline tests/latency_test/tcp_lat.cpp]: @igor-ivanov Please explain reason for this line

**Comment by OphirMunk** [comment]
Merged #202

File: tests/latency_test/tcp_lat.cpp (hunk 1 above)
Comment (igor-ivanov) [inline tests/latency_test/tcp_lat.cpp]: compilation warning

**Pull request closed without merging**